src/test/scala/io/getclump/ClumpSourceSpec.scala

```diff
@@ -62,6 +62,12 @@ class ClumpSourceSpec extends Spec {
     }
   }
 
+  "allows to create a clump source with key function and parameter (ClumpSource.apply)" in {
+      def fetch(session: Long, inputs: Set[Int]) = Future.value(inputs.map(_.toString))
+      val source = Clump.sourceWithParam[Set[Int]].apply[Int, String, (Long, Set[Int])]({ case ((session, inputs)) => fetch(session, inputs)})(_.toInt)
+      clumpResult(source((2L, _)).get(1)) mustEqual Some("1")
```

Comment:
I wanted this to look like:
```scala
def fetch(session: Long, inputs: Set[Int]) = Future.value(inputs.map(_.toString))
val source = Clump.sourceWithParam(fetch)(_.toInt)
clumpResult(source(2L, _).get(1)) mustEqual Some("1")
```

so in a class it might look like
```scala
class EnrichedTracksRepository(tracks: ClumpSourceWithParam[Int, Track, Set[Int], (Session, Set[Int]), ...)
...
getTracks(session: Session, trackIds: Set[Int]) = {
   Clump.traverse(trackIds) { trackId =>
      track <- tracks(session, _).get(trackId)
      user <- users.get(track.userId)
   } yield EnrichedTrack(track, user)
}
```

Maybe the type noise after `ClumpSourceWithParam` makes it not worth it anyway - it's quite ugly. 

Can you think of a more elegant way?